src/net.cpp

```diff
@@ -1895,18 +1903,34 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
         if (interruptNet)
             return;
 
-        // Add seed nodes if DNS seeds are all down (an infrastructure attack?).
-        // Note that we only do this if we started with an empty peers.dat,
-        // (in which case we will query DNS seeds immediately) *and* the DNS
-        // seeds have not returned any results.
-        if (addrman.size() == 0 && (GetTime() - nStart > 60)) {
-            static bool done = false;
-            if (!done) {
-                LogPrintf("Adding fixed seed nodes as DNS doesn't seem to be available.\n");
+        // Static bool makes sure we only add fixed seeds once in binary lifetime.
+        static bool fixedSeedsDone = disableFixedSeeds;
```

Comment:
I like this renaming of the variable `done` to `fixedSeedsDone`, but maybe we can take the opportunity to remove this `static`? Static variables are scope-limited global variables (in general, globals should be avoided). 
<details>
<summary>suggestion</summary>

```diff
-    const bool disableFixedSeeds = !gArgs.GetBoolArg("-fixedseeds", DEFAULT_FIXEDSEEDS);
+    bool addFixedSeeds = gArgs.GetBoolArg("-fixedseeds", DEFAULT_FIXEDSEEDS);
 
-    if (disableFixedSeeds) {
+    if (!addFixedSeeds) {
         LogPrintf("Fixed seeds are disabled\n");
     }
 
     while (!interruptNet)
     {
         ProcessAddrFetch();
 
         if (!interruptNet.sleep_for(std::chrono::milliseconds(500)))
             return;
 
         CSemaphoreGrant grant(*semOutbound);
         if (interruptNet)
             return;
 
         // Static bool makes sure we only add fixed seeds once in binary lifetime.
-        static bool fixedSeedsDone = disableFixedSeeds;
-        if (!fixedSeedsDone && addrman.size() == 0) {
+        if (addFixedSeeds && addrman.size() == 0) {
             // When the node starts with an empty peers.dat, there are a few other sources of peers before
             // we fallback on to fixed seeds: -dnsseed, -seednode, -addnode
             // If none of those are available, we fallback on to fixed seeds immediately, else we allow
             // 60 seconds for any of those sources to populate addrman.
-            bool addFixedSeeds = false;
+            bool addFixedSeedsNow = false;
             // It is cheapest to check if enough time has passed first.
             if (GetTime<std::chrono::seconds>() > nStart + std::chrono::minutes{1}) {
-                addFixedSeeds = true;
+                addFixedSeedsNow = true;
                 LogPrintf("Adding fixed seeds as 60 seconds have passed and addrman is empty\n");
             }
 
             // Checking !dnsseed is cheaper before locking 2 mutexes.
-            if (!addFixedSeeds && !dnsseed) {
+            if (!addFixedSeedsNow && !dnsseed) {
                 LOCK2(m_addr_fetches_mutex, cs_vAddedNodes);
                 if (m_addr_fetches.empty() && vAddedNodes.empty()) {
-                    addFixedSeeds = true;
+                    addFixedSeedsNow = true;
                     LogPrintf("Adding fixed seeds as -dnsseed=0, -addnode is not provided and and all -seednode(s) attempted\n");
                 }
             }
 
-            if (addFixedSeeds) {
+            if (addFixedSeedsNow) {
                 CNetAddr local;
                 local.SetInternal("fixedseeds");
                 addrman.Add(convertSeed6(Params().FixedSeeds()), local);
-                fixedSeedsDone = true;
+                addFixedSeeds = false;
             }
         }
```
</details>

Should new variables be written in `snake_case`?

Reply:
Done! I too prefer the elimination of the static variable. Wasn't sure reviewers would be keen on it, but I agree it is cleaner.